### PR TITLE
feat(CX-2511): add last 3 recent auction results based on artists you collect

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [CX-2511]

### Description

Added a query to get 3 recent auction results. 
When auction results are present, display them in the `<AuctionResultListItemFragmentContainer />` container. 
When pressing on one auction item, navigate to the auction page.

The error on the recording was introduced earlier 🥲
I'll look if we already have a ticket for this.

https://user-images.githubusercontent.com/36167539/167369601-bc69ed28-2bcf-4a03-827c-447ee2ac3c47.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [x] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add last 3 recent auction results based on artists you collect (My Collection Insights) - daria

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look on our [docs], or get in touch with us.

[app state migration]: /docs/adding_state_migrations.md
[feature flag]: /docs/developing_a_feature.md
[docs]: /docs/README.md

[CX-2511]: https://artsyproduct.atlassian.net/browse/CX-2511?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsBasedOnArtistsYouCollect.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        data={auctionResults}
+        listKey="artist-auction-results"
+        renderItem={({ item }) => (
+          <>
+            <AuctionResultListItemFragmentContainer
+              auctionResult={item}
+              showArtistName
+              onPress={() => navigate(`/artist/${item.artistID}/auction-result/${item.internalID}`)}
+            />
+          </>
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsBasedOnArtistsYouCollect.tsx:43]: **question:** is there a reason for the fragment containers `<> </>` here?

**Review by MrSltun — APPROVED** [review]
Great work Daria! 👏 👏  I just left a small comment

File: src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsBasedOnArtistsYouCollect.tsx (hunk 1 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Insights/AuctionResultsBasedOnArtistsYouCollect.tsx:43]: Thank you @MrSltun 🙏 
Oh, nope, the empty tag should go) Thanks for spotting it

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 2]
```diff
@@ -1,25 +1,79 @@
...
       <MyCollectionInsightsOverview />
+      <AuctionResultsBasedOnArtistsYouCollect me={me} />
     </StickyTabPageScrollView>
   )
 }
 
 export const MyCollectionInsightsQR: React.FC<{}> = () => (
   <Suspense fallback={<MyCollectionInsightsPlaceHolder />}>
-    <MyCollectionInsights />
+    {/* <MyCollectionInsights /> */}
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:28]: 👀

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 3]
```diff
@@ -1,25 +1,79 @@
...
+          }
+        }
+      }
+    }
+  `,
+})
+
+export const MyCollectionInsightsQueryRenderer = () => {
+  return (
+    <QueryRenderer<MyCollectionInsightsQuery>
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:59]: Even though QueryRenderer is still used in a lot of components we use [useLazeLoadQuery](https://relay.dev/docs/api-reference/use-lazy-load-query/) for all new screens and components.

Comment (olerichter00) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:59]: Here is the RFC about it https://github.com/artsy/eigen/issues/6080

Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx]: @olerichter00 Done 👍

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 4]
```diff
@@ -18,8 +25,17 @@ export const MyCollectionInsightsQR: React.FC<{}> = () => (
   </Suspense>
 )
 
-const MyCollectionInsightsPlaceHolder = () => (
+// TODO: fix, placeHolder is hidden behind the header
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:28]: Is this something we could fix now or do we need to wait until the new header implementation is done?

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tests.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        }
+      `,
+      {}
+    )
+    return (
+      <StickyTabPage
+        tabs={[
+          {
+            title: "test",
+            content: <AuctionResultsBasedOnArtistsYouCollect auctionResults={queryData.me!} />,
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tests.tsx:38]: Would this work as well?

```suggestion
            content: <AuctionResultsBasedOnArtistsYouCollect auctionResults={queryData.me} />,
```

**Review by olerichter00 — APPROVED** [review]
Thanks a lot for the rewrite 🌟

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx (hunk 4 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:28]: It's a global issue. You can find it on very screen that has tabs

**Pull request merged**